### ISSUE TITLE
Feature: show notifications on active screen

### ISSUE THREAD

**Issue by LordMZTE** [Toqozz/wired-notify#60]
I think it would be neat if wired would support showing notifications on the current active monitor (so the monitor the active window is on) instead of choosing the monitor number.

**Comment by Toqozz** [comment]
Nice idea, past me actually put things in place to allow this in future, so it should be relatively easy to add.

Having the notifications move to the active monitor might be a bit more tricky.

For config, how's setting `monitor: -1` to make it appear on the active one?

**Comment by LordMZTE** [comment]
Sounds like a good idea for the config format!

I think it would actually be nice if the notifications didn't move once they appear on the active monitor. I think they'd be harder to read if they kept jumping around, so I wouldn't even worry about that.

**Comment by Toqozz** [comment]
I've added basic support for this in master.  You can try it by setting `monitor: -1` in your `NotificationBlock` config.

Finding the "active monitor" in x11 is actually surprisingly challenging.

There are basically 2 ways that "monitor focus" can be defined: active window, and cursor.

Active window means that the active monitor is determined based on whatever window is considered to be in focus, whereas cursor is obviously whatever monitor the cursor is on.

For both of these x11 doesn't actually have a decent way of just asking for the active monitor.  Actually multi-monitor in x11 is just super confusing in general since most people use a single x11 "screen" glued together.  To determine the active monitor I need to first find the pixel coordinates of either the active window or cursor, then loop through winit's `Window::available_monitors()` iterator (I think it gets this through xrandr) and check if those coordinates are within that.  o lawd.

On top of that there's a couple choices for how we want to actually handle showing notifications on the active monitor:
1. Windows spawn on the active monitor and stay there, even if focus changes.  This requires a different "stack" of notifications for each monitor, which is kind of annoying.
2. Windows spawn on the active monitor and follow the mouse (actually easier, as it turns out).  I think dunst does this.

I think 1. is the most natural and probably what is wanted, but I'll need to think about it some more.  For now I've implemented 2. with focus determined by the cursor.

In future I think we probably need a deeper configuration here, for specifying "focus follows mouse", "focus follows cursor", and *maybe* the behavior.  It would probably replace the `monitor` field for `NotificationBlock`.

**Comment by Toqozz** [comment]
We now have Mouse/Window focus following in master.

To change the focus follow mode, in your root config (not on a block) do `focus_follows: <Mode>` where `Mode` is either `Mouse` or `Window`:

```rust
(
    max_notifications: 0,
    timeout: 10000,
    poll_interval: 16,
    focus_follows: Mouse,
    ...
)
```

This is still using option 2. above.  Also the option is only changeable globally.  I figure that the use cases for changing this per-notification are tiny and probably not worth the hassle, and have similar ideas about making notifications "append" to each monitor when focus changes, but I still like the idea.  It's just hard to implement :)

Anyway let me know if you find any issues.

**Comment by LordMZTE** [comment]
I just got around to trying this, but it doesn't seem to work. [This](https://mzte.de/git/LordMZTE/dotfiles/src/commit/9a029efec2b1249c3bb1ffbc9a4b9ca9ce6c3e0b/.config/wired/wired.ron) is my wired config. Wired doesn't print any interesting output. Looks like using `focus_follows: Mouse` doesn't work either :/

OS: Arch Linux
WM: i3
wired version: commit 2659b80 (newest one at the time of writing)

**Comment by Toqozz** [comment]
You need `monitor: -1` in your `NotificationBlock`, not the root config, dummy.

**Comment by LordMZTE** [comment]
Whoops, my bad. Works perfectly now!

**Comment by Toqozz** [comment]
Available in [0.10.2](https://github.com/Toqozz/wired-notify/releases/tag/0.10.2).
